experimental/errorsource/error_source_middleware.go

```diff
@@ -0,0 +1,33 @@
+package errorsource
+
+import (
+	"net/http"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/httpclient"
+)
+
+// Error captures error source and implements the error interface
+type Error struct {
+	Source backend.ErrorSource
+
+	Err error
+}
+
+func (r Error) Error() string {
+	return r.Err.Error()
+}
+
+// Middleware captures error source metric
+func Middleware(plugin string) httpclient.Middleware {
+	return httpclient.NamedMiddlewareFunc(plugin, func(opts httpclient.Options, next http.RoundTripper) http.RoundTripper {
+		return httpclient.RoundTripperFunc(func(req *http.Request) (*http.Response, error) {
+			res, err := next.RoundTrip(req)
+			if res != nil && res.StatusCode >= 400 {
```

Comment:
Nit. I'm not sure this condition will ever be true.Based on what I know there will be an error or else a response

Reply:
Sorry my bad I totally read this wrong please disregard :cry: